src/Keboola/StorageApi/Client.php

```diff
@@ -2232,4 +2261,15 @@ public function isAwsDebug()
     {
         return $this->awsDebug;
     }
+
+    /**
+     * @return \GuzzleHttp\Client|null
+     */
+    public function getBaseMockClient()
```

Comment:
this kind of things like `getBaseMockClient`,  `mockResponses` shouldn't be in client.

Reply:
You should be able to inject client into constructor instead for example.

Reply:
Right I agree, I just did it like this to get conversation going :-)

ideally imho a mock client would extend the sapi client, but in order to do that the initialisation of the sapi client would need to change quite fundamentally.

Reply:
yes we can change it

Reply:
but no extensions of sapi client, it should be final.

Reply:
ok.  well from inspection, the feature we need won't require anything like that so we can put it off till another time.